src/main/java/uk/gov/hmcts/divorce/caseworker/event/CaseworkerNoticeOfChange.java

```diff
@@ -132,7 +132,33 @@ public void configure(final ConfigBuilder<CaseData, State, UserRole> configBuild
         caseAccessService.removeUsersWithRole(details.getId(), roles);
 
         return AboutToStartOrSubmitResponse.<CaseData, State>builder()
-            .data(data)
+            .data(correctRepresentationDetails(data, beforeDetails.getData()))
             .build();
     }
+
+    /** On NOC event, CCD is somehow removing solicitor details for the applicant other than the one selected for NOC.
+    * Hence, putting the solicitor details back to the new case details using the before details.
+    * */
+    private CaseData correctRepresentationDetails(final CaseData data,
+                                                  final CaseData beforeData) {
+        Applicant applicantToBeCorrected;
```

Comment:
I don't think this variable is necessary you can just do:

if (data.getNoticeOfChange().getWhichApplicant().equals(APPLICANT_1)) {
  data.getApplicant2().setSolicitor(beforeData.getApplicant2().getSolicitor());
} else {
...
}